docs/authentication/social-connections/microsoft.mdx

```diff
@@ -1,115 +1,128 @@
+1. Fill out the form as follows:
+    1. Under **Name**, name the application whatever you'd like. "Clerk Demo App", for example.
+    1. Under **Supported account types**, select **Accounts in any organizational directory (Any Microsoft Entra ID tenant – Multitenant) and personal Microsoft accounts (e.g. Skype, Xbox)**.
+    1. Under **Redirect URI (Optional)**, select **Web**.
+    1. Finally, select **Register** to submit the form.
 
-Microsoft Azure supports 2 kinds of credentials:
+### Get your client ID and client secret
 
-1. Client secrets (as the majority of oauth providers)
-2. Certificates
+Once your Microsoft Entra ID app is created, or once you select your app from the Entra ID dashboard, you'll be taken to its **Overview**.
 
-At the time of writing, Clerk supports only Client secrets.
+1. From your app's overview, copy the **Application (client) ID** and save it somewhere secure. It's required connecting your Microsoft Entra ID app to your Clerk application.
+1. On this same page, under **Client credentials**, select **Add a certificate or secret** to generate a Client Secret.
+1. You'll be taken to the **Certificate & secrets** tab, where you can select the **New client secret** button. In the modal that appears, add a description and set an expiration time for your secret.
+    > [!IMPORTANT]
+    > When your secret expires, your social connection will stop working until you generate a new client secret and add it to your Clerk app.
+1. Copy the test in your new client secret's **Value** field and save it somewhere secure. You'll add it to your Clerk application later, alongside your client ID.
 
-To create a new client secret, click on the New client secret button as illustrated below:
+### Connect Entra ID app and get your redirect URI
 
-![Certificates & secrets screen](/docs/images/authentication-providers/microsoft/c38a13ba0630dd41a950b75a51f025a34cd9fbea-3360x1782.png)
+1. In another tab, visit the Clerk Dashboard.
+1. In the sidebar, go to **User & Authentication > [Social Connections](https://dashboard.clerk.com/last-active?path=user-authentication/social-connections)**. In the list of **Auth Providers**, enable **Microsoft**.
+1. Ensure that both **Enable for sign-up and sign-in** and **Use custom credentials** are toggled on. Then:
+    - Under **Client ID**, add the value you copied from **Application (client) ID** in the Entra ID dashboard.
+    - Under **Client Secret**, add the client secret value you generated in the Entra ID dashboard.
+    - Copy the **Authorized redirect URI**. You need it for the final step to configure your Entra ID app.
+    - Select **Save**.
 
-You will be prompted to provide a name and lifetime for the new secret.
+### Enable OpenID
 
-![Adding a name and selecting expiration date](/docs/images/authentication-providers/microsoft/08f6445903a233c89ad96be25de3293def891cb6-1172x854.png)
+For OAuth to function, Clerk requires an `id_token` from Microsoft Entra ID.

```

Comment:
We mention `id_token` here but the user doesn't use this, do they? I'm not sure why we mention it.

We can just say something like "To connect your Clerk app to your Microsoft app, you must set the **Authorized redirect URI** in your Microsoft dashboard.